src/TapeRoller.ts

```diff
@@ -0,0 +1,253 @@
+import rimraf from 'rimraf';
+import mkdirp from 'mkdirp';
+import glob from 'glob';
+
+export type ConfigType = Partial<{
+  sourceDirectory: string,
+  targetDirectory: string,
+  sourceFile: string,
+  targetFile: string,
+}>;
+
+export type ModifyType = Partial<{
+  append: string,
+  prepend: string,
+  replace: string,
+  custom: Function,
+}>;
+
+export type StreamType = fs.ReadStream | fs.WriteStream;
+
+export class TapeRoller {
+  private readonly sourceDirectory?: string;
+
+  private sourceFile?: string;
+
+  private readonly targetDirectory?: string;
+
+  private targetFile?: string;
+
+  private stream?: StreamType;
+
+  constructor ({ sourceDirectory, targetDirectory, sourceFile, targetFile }: ConfigType = {}) {
+    this.sourceDirectory = sourceDirectory;
+    this.targetDirectory = targetDirectory;
+    this.sourceFile      = sourceFile;
+    this.targetFile      = targetFile;
+  }
+
+  private static resolvePath (dir?: string, file?: string): string {
+    return resolve(process.cwd(), dir || '', file || '');
+  }
+
+  public static async fromGlob (pattern: string, options?: { [key: string]: any }, targetDirectory?: string): Promise<TapeRoller[]> {
+    const asyncGlob = promisify(glob);
+    const files     = await asyncGlob(pattern, options);
+
+    return files.map(sourceFile => new TapeRoller({ targetDirectory, sourceFile }));
+  }
+
+  public rename (oldPath: string, newPath: string): this {
+    fs.renameSync(oldPath, newPath);
+
+    return this;
+  }
+
+  public read (file?: string): this {
+    if (file) {
+      this.sourceFile = file;
+    }
+
+    if (!this.sourceFile) {
+      throw new Error('No source file provided');
+    }
+
+    const path = TapeRoller.resolvePath(this.sourceDirectory, this.sourceFile);
+
+    this.stream = fs.createReadStream(path);
+
+    this.stream.on('error', (error) => console.error('Read error: ', error));
+
+    return this;
+  }
+
+  public write (file?: string, temp?: boolean): this {
+    if (file) {
+      this.targetFile = file;
+    }
+
+    const fileName = temp ? `__temp__${this.targetFile}` : this.targetFile;
+    const path     = TapeRoller.resolvePath(this.targetDirectory);
+    const fullPath = `${path}/${fileName}`;
+
+    mkdirp.sync(this.targetDirectory);
+
+    this.stream = this.stream.pipe(fs.createWriteStream(fullPath));
+
+    this.stream.on('error', (error) => console.error('Write error: ', error));
+
+    if (temp) {
+      this.stream.on('close', () => {
+        this.rename(fullPath, `${path}/${this.targetFile}`);
+      });
+    }
+
+    return this;
+  }
+
+  public copy (source?: string, target?: string): this {
+    if (source) {
+      this.sourceFile = source;
+    }
+
+    if (target) {
+      this.targetFile = target;
+    }
+
+    if (!this.stream) {
+      this.read();
+    }
+
+    this.write();
+
+    return this;
+  }
+
+  public replace (parameters: { [key: string]: any }) {
+    if (!this.stream) {
+      this.read();
```

Comment:
Is this a good idea? If there's no file set you should maybe just throw a new Error. Make read something that's required (modify can call read itself).